### PR TITLE
Show notifications! on Android, in foreground

### PR DESCRIPTION
**Pull request by gnprice**
This implements most of the remaining work of #320.

It doesn't yet deliver most of the user-facing value: for that, the
remaining piece will be to make notifications appear when the app is
in the background as well as when it's in the foreground.  I have a
draft for that and it works, but the tests require some more fiddling
which I may not get to today, so I wanted to get the bulk of this
posted for review.

Many other improvements still to be made are tracked as their own
issues: most notably iOS support #321, and opening a conversation
by tapping the notification #123.

### CODE REVIEW

**Review by chrisbobbe — APPROVED** [review]
Wow! I tested this out on the office Android device and it worked as expected.

A few small points below, then please merge at will.

File: ios/Podfile.lock [hunk 1]
```diff
@@ -70,6 +70,8 @@ PODS:
     - GoogleUtilities/UserDefaults (~> 7.8)
     - nanopb (< 2.30910.0, >= 2.30908.0)
   - Flutter (1.0.0)
+  - flutter_local_notifications (0.0.1):
```
[the remark this answers is not in the document]
Comment (gnprice) [reply ios/Podfile.lock:73]: Yeah, good question. Replied in chat here: https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/notification.20libraries/near/1670983

In short: indeed it does have such baggage, and we'll likely want to move to writing our own thin layer with Pigeon in the future, but this library is what's recommended by Flutter upstream and seems to be the best existing thing to use.

File: lib/notifications.dart [hunk 2]
```diff
@@ -71,9 +74,114 @@ class NotificationService {
...
+      FcmMessageDmRecipient() =>
+        data.senderFullName,
+    };
+    ZulipBinding.instance.notifications.show(
+      kNotificationId,
+      title,
+      data.content,
+      NotificationDetails(android: AndroidNotificationDetails(
+        NotificationChannelManager.kChannelId,
+        '(Zulip internal error)', // TODO never implicitly create channel: https://github.com/MaikuB/flutter_local_notifications/issues/2135
```
[the remark this answers is not in the document]
Comment (gnprice) [reply lib/notifications.dart:165]: Yeah, exactly.

File: ios/Podfile.lock (hunk 1 above)
[the remark this answers is not in the document]
Comment (gnprice) [reply ios/Podfile.lock:73]: Filed that thought as #351.

File: lib/notifications.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (gnprice) [reply lib/notifications.dart:172]: Added a comment explaining this more.

**Comment by gnprice** [comment]
Thanks for the review! Revised to add a comment, and merged.

**Pull request merged**